scripts/controller.py

```diff
@@ -132,16 +155,16 @@
 
     # Write controller CMakeLists.txt
     with open(os.path.join(controller_path, "CMakeLists.txt"), "w") as f:
-        f.write(cmake_template.format(controller_name=controller_name))
+        f.write(cmake_template.format(controller=controller))
 
     # Write controller source file
-    with open(os.path.join(controller_path, f"{controller_name}.cpp"), "w") as f:
+    with open(os.path.join(controller_path, f"{controller}.cpp"), "w") as f:
         f.write(controller_template)
 
     # Add controller directory to the controllers CMakeList.txt
     with open(os.path.join(controller_path, "..", "CMakeLists.txt"), "a") as f:
-        f.write(f"add_subdirectory({controller_name})\n")
+        f.write(f"add_subdirectory({controller})\n")
 
     # Append new controller binary to .gitignore
     with open(".gitignore", "a") as f:
-        f.write("{}\n".format(os.path.join(os.sep, controller_path, controller_name)))
+        f.write("{}\n".format(os.path.join(os.sep, os.path.relpath(controller_path, b.project_dir), controller)))
```

Comment:
It's probably worth changing the structure of the gitignore so that the lines this adds are grouped with the binaries.